### PR TITLE
Ex3 automated testing

### PR DESCRIPTION
**Pull request by thebauSoftwire**
Adding unit tests and integration test to app

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,4 +58,19 @@ Press CTRL+C to quit
  * Debugger is active!
  * Debugger PIN: 113-666-066
 ```
+
 Now visit [`http://localhost:5000/`](http://localhost:5000/) in your web browser to view the app.
+
+## Testing
```
Comment (JackMeadDev) [inline README.md:64]: Thanks for the clear simple instructions 👍

File: todo_app/app.py [hunk 2]
```diff
@@ -1,24 +1,29 @@
...
-app = Flask(__name__)
-app.config.from_object(Config())
+def create_app():
+    app = Flask(__name__)
+    app.config.from_object(Config())
 
+    @app.route('/')
+    def index():
+        todoItems = get_items()
+        app.logger.info(todoItems)
```
Comment (JackMeadDev) [inline todo_app/app.py]: This looks like pretty useless output in my terminal - it just looks like an array with a series of `<todo_app.data.item.Item object at address>`!

We can make that a bit nicer with the magic `__repr__` function on a class if we want to, for example inside your `Item` class if we add:
```python
    def __repr__(self):
        return f'Ticket {self.id} is in status {self.status}'
```
Then we'll get much more human readable logging here. You'll notice I've avoided adding any reference to the Item name - very intentionally! That could easily end up with some personal data (e.g. "take <person> to see GP about <medical condition>" or "send email to the king at <private email address>" or something!) and we generally want to keep that out of our logging, so we can confidently share logs with people that we wouldn't necessarily give production database access to for example

File: todo_app/templates/index.html [hunk 3]
```diff
@@ -1,53 +1,46 @@
...
-        name="todoItemTitle"
-        placeholder="Enter task"
-        class="form-control"
-      />
+    <div class="col">
+      <h2>Completed Items</h2>
+      <ul class="list-group mb-4">
+        {% for todoItem in view_model.done_items %}
+        <li class="list-group-item d-flex justify-content-between align-items-center">
+          <div class="d-flex align-items-center">
```
Comment (JackMeadDev) [inline todo_app/templates/index.html:37]: Nice work on the CSS, it looks great and a lot shinier than most I get to see!😄

File: todo_app/view_models/test_todo_list_view_model.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+from todo_app.data.item import Item
+
+test_items = [
+    Item(1, 'some-name','To Do'),
+    Item(2, 'some-name','Done'),
+    Item(3, 'some-name','To Do'),
+    Item(4, 'some-name','Done'),
+]
+
+def test_view_model_sets_items():
```
Comment (JackMeadDev) [inline todo_app/view_models/test_todo_list_view_model.py]: Very minor but one to consider; there's a small risk here that somebody might modify this `test_items` list during a test, which could impact later tests if run in sequence. For that reason, we might potentially consider using a fixture that generates a fresh set of items per test:
```python
@pytest.fixture
def test_items():
    return [
        Item(1, 'some-name','To Do'),
        Item(2, 'some-name','Done'),
        Item(3, 'some-name','To Do'),
        Item(4, 'some-name','Done'),
    ]

def test_view_model_sets_items(test_items):
```

File: todo_app/view_models/todo_list_view_model.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+        self._items = items
+
+    @property
+    def items(self):
+        return self._items
+    
+    @property
+    def done_items(self):
+        print(self.items)
+        complete_items = filter(lambda item: item.status == 'Done', self.items)
```
Comment (JackMeadDev) [inline todo_app/view_models/todo_list_view_model.py:12]: Nice concise solution here 👍

**Review by JackMeadDev — APPROVED** [review]
Nice work, I was easily able to run your code & tests and you've got simple effective approaches to both levels of testing 😄 I've left a couple of thoughts for consideration, but go ahead and merge when you're ready!

File: todo_app/app.py (hunk 2 above)
Comment (thebauSoftwire) [reply todo_app/app.py]: ahh I accidentally left this in when debugging but this is really helpful information! Thanks :))

**Pull request merged**